src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -63,63 +62,116 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
 
-                Debug.Assert(!newTypeSyntax.ContainsDiagnostics, "Explicit type replacement likely introduced an error in code");
+                var tupleDeclaration = GenerateTupleDeclaration(tupleTypeSymbol, variableDesignation).WithLeadingTrivia(leadingTrivia);
 
-                editor.ReplaceNode(typeSyntax, newTypeSyntax);
+                editor.ReplaceNode(declarationExpression, tupleDeclaration);
             }
             else
             {
-                var tupleTypeSymbol = semanticModel.GetTypeInfo(typeSyntax.Parent, cancellationToken).ConvertedType;
+                var typeSymbol = semanticModel.GetTypeInfo(typeSyntax, cancellationToken).ConvertedType;
+                RoslynDebug.AssertNotNull(typeSymbol);
 
-                var leadingTrivia = node.GetLeadingTrivia()
-                    .Concat(parensDesignation.GetAllPrecedingTriviaToPreviousToken().Where(t => !t.IsWhitespace()).Select(t => t.WithoutAnnotations(SyntaxAnnotation.ElasticAnnotation)));
+                editor.ReplaceNode(typeSyntax, GenerateTypeDeclaration(typeSyntax, typeSymbol));
+            }
+        }
+
+        private static Task HandleForEachStatementAsync(Document document, SyntaxEditor editor, ForEachStatementSyntax forEach, CancellationToken cancellationToken)
+            => UpdateTypeSyntaxAsync(
+                document,
+                editor,
+                forEach.Type,
+                forEach.Identifier.GetRequiredParent(),
+                cancellationToken);
+
+        private static Task HandleVariableDeclarationAsync(Document document, SyntaxEditor editor, VariableDeclarationSyntax varDecl, CancellationToken cancellationToken)
+            => UpdateTypeSyntaxAsync(
+                document,
+                editor,
+                varDecl.Type,
+                // Since we're only dealing with variable declaration using var, we know
+                // that implicitly typed variables cannot have multiple declarators in
+                // a single declaration (CS0819). Only one variable should be present
+                varDecl.Variables.Single().Identifier.Parent!,
+                cancellationToken);
+
+        private static async Task UpdateTypeSyntaxAsync(Document document, SyntaxEditor editor, TypeSyntax typeSyntax, SyntaxNode declarationSyntax, CancellationToken cancellationToken)
+        {
+            typeSyntax = typeSyntax.StripRefIfNeeded();
 
-                var tupleDeclaration = GenerateTupleDeclaration(tupleTypeSymbol, parensDesignation).WithLeadingTrivia(leadingTrivia);
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var typeSymbol = semanticModel.GetTypeInfo(typeSyntax, cancellationToken).ConvertedType;
+            RoslynDebug.AssertNotNull(typeSymbol);
```

Comment:
how do we know this is safe?

Reply:
What case does `GetTypeInfo` fail on `TypeSyntax`? I thought if we're guaranteed the `SyntaxNode` then it is exceptional if this fails.

Reply:
@ryzngard Here is a compiler test showing that it *may* return null:

https://github.com/dotnet/roslyn/blob/30ed50e62601a63c86c58ff1830936f778b328c1/src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelGetSemanticInfoTests.cs#L405

I don't know if this scenario is relevant for this feature, but there could be other scenarios where it can return null.

Reply:
I think the only time it would be null is in cases where there is no conversion, and var always fills in `ConvertedType` information. That said, I'm happy to add telemetry for if we see this start to fail we can adjust accordingly. This would have failed prior on line 112 or 102. NRE would not be new here.

Reply:
i just don't believe this is safe.  i don't see any reason why ConvertedType would always be known to be non-null.  It definitely isn't doc'ed as that.

Reply:
I'm not saying `ConvertedType` is safe, I'm saying if there is a gap in assumption of safety at this particular line it was already assumed. The code prior to this would have thrown an NRE in exactly the same way. I can add a bail here if that's preferred but that might be even worse by covering up an issue we assumed wasn't possible prior to `#enable nullable` on this file

Reply:
gotcha.  yes. please be resilient to this being null.  the old code was bad (not your fault).  i think we should be careful and trust NRT here.